examples/common_vmseries_and_autoscale/main.tf

```diff
@@ -0,0 +1,239 @@
+  existing_pip_prefix_resource_group_name = try(each.value.existing_pip_prefix_resource_group_name, null)
+
+
+  tags       = var.tags
+  depends_on = [module.vnet]
+}
+
+
+
+# create load balancers, both internal and external
+module "load_balancer" {
+  source = "../../modules/loadbalancer"
+
+  for_each = var.load_balancers
+
+  name                = "${var.name_prefix}${each.value.name}"
+  location            = var.location
+  resource_group_name = local.resource_group.name
+  enable_zones        = var.enable_zones
+  avzones             = try(each.value.avzones, null)
+
+  network_security_group_name          = try(each.value.network_security_group_name, null)
+  network_security_resource_group_name = try(each.value.network_security_group_rg_name, null)
+  network_security_allow_source_ips    = try(each.value.network_security_allow_source_ips, [])
+
+  frontend_ips = {
+    for k, v in each.value.frontend_ips : k => {
+      create_public_ip         = try(v.create_public_ip, false)
+      public_ip_name           = try(v.public_ip_name, null)
+      public_ip_resource_group = try(v.public_ip_resource_group, null)
+      private_ip_address       = try(v.private_ip_address, null)
+      subnet_id                = try(module.vnet[v.vnet_key].subnet_ids[v.subnet_key], null)
+      in_rules                 = try(v.in_rules, {})
+      out_rules                = try(v.out_rules, {})
+    }
+  }
+
+  tags       = var.tags
+  depends_on = [module.vnet]
+}
+
+
+# Create the scale sets and related resources.
+module "ai" {
+  source = "../../modules/application_insights"
+
+  for_each = { for k, v in var.vmss : k => "${v.name}-ai" if can(v.autoscale_metrics) }
+
+  name                = "${var.name_prefix}${each.value}"
+  resource_group_name = local.resource_group.name
+  location            = var.location
+
+  workspace_mode            = try(var.application_insights.workspace_mode, null)
+  workspace_name            = try(var.application_insights.workspace_name, "${var.name_prefix}${each.key}-wrkspc")
+  workspace_sku             = try(var.application_insights.workspace_sku, null)
+  metrics_retention_in_days = try(var.application_insights.metrics_retention_in_days, null)
+
+  tags = var.tags
+}
+
+module "appgw" {
+  source = "../../modules/appgw"
+
+  for_each = var.appgws
+
+  name                = "${var.name_prefix}${each.value.name}"
+  resource_group_name = local.resource_group.name
+  location            = var.location
+  subnet_id           = module.vnet[each.value.vnet_key].subnet_ids[each.value.subnet_key]
+
+  managed_identities = try(each.value.managed_identities, null)
+  waf_enabled        = try(each.value.waf_enabled, false)
+  capacity           = try(each.value.capacity, null)
+  capacity_min       = try(each.value.capacity_min, null)
+  capacity_max       = try(each.value.capacity_max, null)
+  enable_http2       = try(each.value.enable_http2, null)
+  zones              = var.enable_zones ? try(each.value.zones, null) : null
```

Comment:
Shall we add this `var.enable_zones` condition to other examples as well? It doesn't check for the enable_zones variable in others.

Reply:
I would keep keep it as it is and start to create a list of things to refactor. We already have #280 and #279 that will make a huge change in the code. Maybe we can move all three to a single issue?

Reply:
Converted it back to match with the other examples. Put a comment on #280 for this.